crawl-ref/source/defines.h

```diff
@@ -194,12 +194,16 @@ const int AGILITY_BONUS = 5;
 
 #define MAX_MONSTER_HP 10000
 
+// Many, MANY places currently hard-code this to 8 bits, but we need to
+// expand it. Please use colour_t in new code.
+typedef uint8_t colour_t;
+
 // colours, such pretty colours ...
 // The order is important (IRGB bit patterns).
-enum COLOURS
+enum COLOURS : colour_t
 {
-    COLOUR_INHERIT = -1,
-    BLACK,
+    COLOUR_INHERIT = (colour_t)-1,
```

Comment:
I'm a bit worried about this because COLOUR_INHERIT will no longer test as `< 0`, one would need confidence that its sign is never used (i.e. color values are never tested with `>= 0` -- did you check this? (It's kind of hard to search for unfortunately)